### PR TITLE
💩💩💩💩💩💩💩💩💩💩💩(Transfer passport)💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩💩

### LINKED ISSUE #67: Allow swiping or transferring of a passport
## Scope

- When scanning a QR code that is a private key with the burner wallet, the user is prompted to enter an amount of value from the side chain that they'd like to withdraw
- Users can also send money easily
- In Planet A, there's however also a passport that a user owns. This passport is an NFT
- It's a user's expectation that NFTs are transferrable
- Currently, sending a passport is not implemented and would have to be done manually (by writing some code)

## Deliverables

- Make sending a passport easy for the average user. This can either be:
  - a "transfer passport" dialogue *and/or*
  - allowing the user to "swipe" a passport in the "WithdrawFromPrivate.js" view
- Make a decision on what's easier for the user and implement this approach

## Roles
bounty gardener: @TimDaub / 10%
bounty worker: name / 75%
bounty reviewer: name / 15%

### PR DESCRIPTION
**Pull request by TimDaub**
Fixes #67 

Screen Shot 2019-08-15 at 15 23 32
Screen Shot 2019-08-15 at 15 23 22
Screen Shot 2019-08-15 at 15 06 25

TODOs:

- [x] View for transferring a passport
- [x] Make sure that scanning a QR code address works in the transfer passport view
- [x] Build and test actual passport transfer functionality on Plasma
- [x] Remove unnecessary console.logs
- [x] Get rid of (most) TODOs
- [x] Integrate passport into receipt view
- [x] Move passport data into config? (I created https://github.com/social-dist0rtion-protocol/planet-a/issues/81)
- [ ] Bug: Go to "Transfer Passport" view and then try closing the view with the X, then it's closing the passport.

### CODE REVIEW

File: src/components/SendByScan.js [hunk 1]
```diff
@@ -115,8 +115,8 @@ class SendByScan extends Component {
 
         if (dataSplit.length === 1 && Web3.utils.isAddress(dataSplit[0])) {
           console.log("RETURN STATE:",this.props.returnState)
-          let returnState = this.props.parseAndCleanPath(dataAfterColon)
-          this.props.returnToState(returnState)
+          let returnState = this.props.parseAndCleanMyAss(dataAfterColon)
+          this.props.returnToState(returnState, this.props.returnState.goBackView)
```
Comment (TimDaub) [inline src/components/SendByScan.js:119]: I did `git add -p` and pressed s to split the 💩 (chunk), but git told me

```
Stage this 💩  [y,n,q,a,d,e,?]? s
Sorry, cannot split this 💩 
```

😭

**Comment by vrde** [comment]
Quick comment about the UI: transfer is too prominent, seems like it's an important call to action since it's at the same level as handshake and plant trees.

**Comment by TimDaub** [comment]
> Quick comment about the UI: transfer is too prominent, seems like it's an important call to action since it's at the same level as handshake and plant trees.

Yes, we'll handle in #21

**Review by vrde — APPROVED** [review]
LGTM

**Pull request merged**